rotkehlchen/data/importer.py

```diff
@@ -182,3 +186,183 @@ def import_cointracking_csv(self, filepath: Path) -> None:
+            - DeserializationError if something is wrong with the format of the expected values
+            - UnsupportedCryptocomEntry if importing of this entry is not supported.
+            - IndexError if the CSV file is corrupt
+            - UnknownAsset if one of the assets founds in the entry are not supported
+        """
+        row_type = csv_row['Transaction Kind']
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Timestamp (UTC)'],
+            formatstr='%Y-%m-%d %H:%M:%S',
+            location='crypto.com',
+        )
+        description = csv_row['Transaction Description']
+        notes = f'{description}\nSource: crypto.com (CSV import)'
+
+        # No fees info for now (Aug 2020) on crypto.com, so we put 0 fees
+        fee = Fee(ZERO)
+        fee_currency = A_USD  # whatever (used only if there is no fee)
+
+        if row_type in (
+            'crypto_purchase',
+            'crypto_exchange',
+            'referral_gift',
+            'crypto_earn_interest_paid',
+        ):
+            # variable mapping to raw data
+            currency = csv_row['Currency']
+            to_currency = csv_row['To Currency']
+            native_currency = csv_row['Native Currency']
+            amount = csv_row['Amount']
+            to_amount = csv_row['To Amount']
+            native_amount = csv_row['Native Amount']
+
+            trade_type = TradeType.BUY if to_currency != native_currency else TradeType.SELL
+
+            if row_type in ('crypto_exchange'):
+                # trades crypto to crypto
+                base_asset = Asset(to_currency)
+                quote_asset = Asset(currency)
+                if quote_asset is None:
+                    raise DeserializationError('Got a trade entry with an empty quote asset')
+                base_amount_bought = deserialize_asset_amount(to_amount)
+                quote_amount_sold = deserialize_asset_amount(amount)
+            else:
+                base_asset = Asset(currency)
+                quote_asset = Asset(native_currency)
+                base_amount_bought = deserialize_asset_amount(amount)
+                quote_amount_sold = deserialize_asset_amount(native_amount)
+
+            rate = Price(abs(quote_amount_sold / base_amount_bought))
+            pair = TradePair(f'{base_asset.identifier}_{quote_asset.identifier}')
+            trade = Trade(
+                timestamp=timestamp,
+                location=Location.CRYPTOCOM,
+                pair=pair,
+                trade_type=trade_type,
+                amount=base_amount_bought,
+                rate=rate,
+                fee=fee,
+                fee_currency=fee_currency,
+                link='',
+                notes=notes,
+            )
+            self.db.add_trades([trade])
+
+            # crypto_purchase is buying crypto assets using a card so
+            # it should also be considered as a deposit movement
+            if row_type == 'crypto_purchase':
```

Comment:
I am not sure about this. Why do you want this to be a deposit movement? From the name "Crypto purchase" it sounds to me as if this is actually some form of trade. If you exchange EUR coming from your card to buy BTC then this is definitely a trade. If you use your card to deposit EUR into your crypto.com account then that's an asset movement.

Asset movements are essentially deposits and withdrawals in exchanges.

Reply:
I was considering buying via card as a deposit + trade but I guess it's not necessary here 👍